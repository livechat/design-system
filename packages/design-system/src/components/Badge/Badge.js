import * as React from 'react';
import * as PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './style.scss';

function Badge({ children, className: extraClassName, secondary, ...props }) {
  const className = cx(styles.badge, {
    [styles['badge--secondary']]: secondary,
    ...(extraClassName ? {[extraClassName]: true} : {})
  });

  return (
    <span className={className} {...props}>
      {children}
    </span>
  );
}

Badge.propTypes = {
  children: PropTypes.node.isRequired,
  secondary: PropTypes.bool
};

export default Badge;
