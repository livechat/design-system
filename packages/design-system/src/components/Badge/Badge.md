### Badge

## Usage

Use badges to draw attention errors and new, updated or important content, like unread messages. Hide badhe when the count is 0.

```js
<p>
  <Badge>7</Badge> Tickets
</p>
<p>
  <Badge>!</Badge> Settings
</p>
<p>
  <Badge>99+</Badge> Chats
</p>
```

## Badge placement

Place badges directly adjoining the related UI object. Badges can appear to the right or left of an UI object

```js
<Badge>!</Badge>
```

## Over 99

If a badge needs to display a number above 99, use “99+”

## Style

Badges are small enough to fit next to an UI object. They have a bold, and filled style that makes them stand out, and a fully rounded border radius.

## Color

- Red reserved to display a sense of urgency or indicate state.
- Light Variation is best for non-essential element on the screen and should recede from the primary content.

# Primary

Errors and important content, like unread messages, new tickets.

```js
<Badge>1</Badge>
<Badge>11</Badge>
<Badge>99+</Badge>
```

Non-essential content like new features or updates.

```js
<Badge secondary>1</Badge>
<Badge secondary>11</Badge>
<Badge secondary>99+</Badge>
```
