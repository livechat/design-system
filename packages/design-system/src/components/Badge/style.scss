@import '../../scss/variables/colors';

.badge {
  display: inline-block;
  box-sizing: border-box;
  padding: 0 3px;
  height: 18px;
  min-width: 18px;

  font-size: 10px;
  font-weight: 600;
  line-height: 18px;
  text-align: center;
  border-radius: 18px;

  & + & {
    margin-left: 4px;
  }

  color: $badge-primary-color;
  background-color: $badge-primary-bg;
}

.badge--secondary {
  color: $badge-secondary-color;
  background-color: $badge-secondary-bg;
}
