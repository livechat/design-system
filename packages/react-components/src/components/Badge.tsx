import * as React from "react";
import cx from "classnames";

export interface IBadgeProps {
  children: React.ReactElement;
  secondary: boolean;
  className?: string;
}

export const Badge = ({
  children,
  className: extraClassName = "",
  secondary = false,
}: IBadgeProps) => {
  const className = cx("lc-badge", {
    "lc-badge--secondary": secondary,
    [extraClassName]: !!extraClassName,
  });

  return <span className={className}>{children}</span>;
};
