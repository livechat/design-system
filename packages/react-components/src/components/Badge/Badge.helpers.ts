export function formatCount(count: number, max: number): string {
  return count > max ? `${max}+` : `${count}`;
}
