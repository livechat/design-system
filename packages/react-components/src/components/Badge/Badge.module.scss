$base-class: 'badge';

@mixin dot($size) {
  border-radius: $size;
  height: $size;
  width: $size;
}

.#{$base-class} {
  align-items: center;
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: column;
  font-size: 12px;
  font-weight: 600;
  gap: 10px;
  justify-content: center;
  line-height: 16px;

  &__dot {
    background-color: var(--content-invert-default);
    margin: 0 -2px;
  }

  & + & {
    margin-left: 4px;
  }

  &--large {
    border-radius: 24px;
    height: 24px;
    min-width: 24px;
    padding: 4px 8.5px;

    .#{$base-class}__dot {
      @include dot(10px);
    }
  }

  &--medium {
    border-radius: 20px;
    height: 20px;
    min-width: 20px;
    padding: 2px 7px;

    .#{$base-class}__dot {
      @include dot(8px);
    }
  }

  &--compact {
    border-radius: 24px;
    height: 18px;
    min-width: 18px;
    padding: 1px 6px;

    .#{$base-class}__dot {
      @include dot(6px);
    }
  }

  &--primary {
    background-color: var(--color-negative-default);
    color: var(--content-invert-default);

    &:hover {
      background-color: var(--color-negative-hover);
    }

    .#{$base-class}__dot {
      background-color: var(--content-invert-default);
    }
  }

  &--secondary {
    background-color: var(--color-action-default);
    color: var(--content-invert-default);

    &:hover {
      background-color: var(--color-action-hover);
    }

    .#{$base-class}__dot {
      background-color: var(--content-invert-default);
    }
  }

  &--tertiary {
    background-color: var(--surface-secondary-default);
    color: var(--content-default);

    &:hover {
      background-color: var(--surface-secondary-hover);
    }

    .#{$base-class}__dot {
      background-color: var(--content-default);
    }
  }
}
