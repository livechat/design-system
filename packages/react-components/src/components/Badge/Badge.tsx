import * as React from 'react';
import cx from 'classnames';

export interface IBadgeProps {
  secondary?: boolean;
  className?: string;
}

export const Badge: React.FC<IBadgeProps> = ({
  children,
  className: extraClassName = '',
  secondary = false,
}) => {
  const className = cx('lc-badge', {
    'lc-badge--secondary': secondary,
    [extraClassName]: !!extraClassName,
  });

  return <span className={className}>{children}</span>;
};
