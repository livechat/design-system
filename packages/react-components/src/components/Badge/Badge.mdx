import { Meta, Title, Subtitle, Canvas, ArgTypes } from '@storybook/blocks';
import { Figma } from '@storybook/addon-designs/blocks';
import * as BadgeStories from './Badge.stories';
import {
  BlockColumnsContainer,
  BlockColumns,
} from '../../stories/components/BlockColumns';

<Meta of={BadgeStories} />

<Title>Badge</Title>

[Component API](#ComponentAPI) | [Content Spec](#ContentSpec)

<Canvas of={BadgeStories.Default} />

#### Example implementation

```jsx
<Badge size="medium" count={1} />
```

## Component API <a id="ComponentAPI" />

<ArgTypes />

## Content Spec <a id="ContentSpec" />

<Figma
  showLink
  height="100%"
  url="https://www.figma.com/file/9FDwjR8lYvincseDkKypC4/Component-Documentations?type=design&node-id=731-59622&mode=dev"
/>
