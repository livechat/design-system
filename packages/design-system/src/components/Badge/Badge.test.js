import * as React from 'react';
import { shallow } from 'enzyme';
import Badge from './Badge';

describe('Badge', () => {
  it('should render primary badge', () => {
    const component = shallow(<Badge>!</Badge>);

    expect(component).toMatchSnapshot();
  });

  it('should render secondary badge', () => {
    const component = shallow(<Badge secondary>!</Badge>);

    expect(component).toMatchSnapshot();
  });
});
