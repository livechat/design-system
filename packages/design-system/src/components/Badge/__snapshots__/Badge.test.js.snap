// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render secondary badge 1`] = `
<span
  className="badge badge--secondary"
>
  !
</span>
`;

exports[`Badge should render primary badge 1`] = `
<span
  className="badge"
>
  !
</span>
`;
